inc/link.class.php

```diff
@@ -83,185 +84,82 @@ public function showItemGenerationForm($params)
+
+                    $row['name'] = $item->fields["name"] ?? "";
+                    $row['otherserial'] = $item->fields["otherserial"] ?? "";
+                    $row['states_id'] = $item->fields["states_id"] ?? "";
+                    $row['locations_id'] = $item->fields["locations_id"] ?? "";
+                    $row['groups_id'] = $item->fields["groups_id"] ?? "";
+                    $row['immo_number'] = $item->fields["immo_number"] ?? "";
+                    $row['template_name'] = $reference->getTemplateName($itemtype, $templateID);
                 } else {
-                    $entity = $_SESSION["glpiactive_entity"];
-                    echo "<input type='hidden' name='id[$i][entities_id]' value="
-                    . $entity . ">";
+                    $row['name'] = false;
+                    $row['otherserial'] = false;
+                    $row['states_id'] = false;
+                    $row['locations_id'] = false;
+                    $row['groups_id'] = false;
+                    $row['immo_number'] = false;
+                    $row['template_name'] = "";
                 }
-                echo "<td>";
 
-                echo "<span id='show_location_by_entity_id_$i'>";
-                Location::dropdown(['name'   => "id[$i][locations_id]",
-                    'entity' => $entity,
-                    'value'  => $locations_id,
-                ]);
-                 echo "</span>";
-                 echo "</td>";
-                 echo "<td>";
-                 echo "<span id='show_group_by_entity_id_$i'>";
-                 Group::dropdown(['name'   => "id[$i][groups_id]",
-                     'entity' => $entity,
-                     'value'  => $groups_id,
-                 ]);
-                 echo "</span>";
-                 echo "</td>";
-                 echo "<td>";
-                 echo "<span id='show_state_by_entity_id_$i'>";
-                 $condition = self::getCondition($itemtype);
-                 State::dropdown(['name'      => "id[$i][states_id]",
-                     'entity'    => $entity,
-                     'condition' => $condition,
-                     'value'     => $states_id,
-                 ]);
-                 echo "</span>";
-                 echo "</td>";
-                 echo "</tr>";
-                 echo Html::hidden("id[$i][id]", ['value' => $key]);
-                 $found = true;
+                $item_rows[] = $row;
+                $found = true;
             }
             $i++;
         }
 
-        if (!$found) {
-            echo "<tr><td align='center' colspan='$colspan' class='tab_bg_2'>" .
-              __("No item to generate", "order") . "</td></tr>";
-        }
-
-        echo "</table>";
-        echo "</div>";
-
         if (!$found) {
             return false;
         }
+
+        // Render the template with all prepared data
+        TemplateRenderer::getInstance()->display('@order/generate_item.html.twig', [
```

Comment:
Template is missing from PR

![image](https://github.com/user-attachments/assets/4d647200-bc5a-4d4d-96fb-5edce5913a30)